app/decorators/provider_decorator.rb

```diff
@@ -69,4 +69,10 @@ def application_defined_fields_data
                         FieldsDefinitionDecorator.new(field).new_application_data(self)
                       end
   end
+
+  protected
+
+  def service_plans_visible
+    h.can?(:manage, :service_contracts) && settings.service_plans.allowed? && settings.service_plans_ui_visible?
```

Comment:
Maybe refactor that and this one https://github.com/3scale/porta/blob/master/app/helpers/vertical_nav_helper.rb#L94
and maybe other places into a helper (though it does not belong to a helper per se)

It will make it easier to refactor later